sdk/storage/azure-storage-file-datalake/azure/storage/file/out/datalake/_models.pyi

```diff
@@ -0,0 +1,93 @@
+# Stubs for datalake._models (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from azure.core.paging import PageIterator
+from azure.storage.blob import AccountSasPermissions, BlobSasPermissions, ContainerProperties, ContainerSasPermissions, ContentSettings, LeaseProperties, ResourceTypes, UserDelegationKey
+from azure.storage.blob._models import ContainerPropertiesPaged
+from azure.storage.blob._shared.models import DictMixin
+from enum import Enum
+from typing import Any, Optional
+
+class FileSystemProperties(ContainerProperties):
+    def __init__(self, **kwargs: Any) -> None: ...
+
+class FileSystemPropertiesPaged(ContainerPropertiesPaged):
+    def __init__(self, *args: Any, **kwargs: Any) -> None: ...
+
+class DirectoryProperties(DictMixin):
```

Comment:
`DirectoryProperties`, `FilePropertie` and `PathProperties` should be merged into a single type. There is almost 100% overlap between them.

Reply:
pending